src/Domain/Donation.php

```diff
@@ -89,7 +89,8 @@ class Donation extends SalesforceWriteProxy
     /**
      * @var string|null PSP's transaction ID assigned on their processing.
      *
-     * In the case of stripe (which is the only thing we support at present, this is the payment intent ID)
+     * In the case of stripe (which is the only thing we support at present, this is the payment intent ID).
+     * May change after initial assignment until donation's Collected, in some edge cases.
```

Comment:
:+1:  Useful comment about stripe.

Not suggesting doing it on this PR but I think I'd support going further and renaming to `stripePaymentIntentId`. If we support any other PSP in future then the transactionIds of that PSP won't really be usable interchangeably with the stripe one so might as well be on a different field.